standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/parser/PartFilterVisitor.java

```diff
@@ -231,25 +229,27 @@ public Long visitIntegerLiteral(PartitionFilterParser.IntegerLiteralContext ctx)
   }
 
   @Override
-  public Date visitDateLiteral(PartitionFilterParser.DateLiteralContext ctx) {
+  public String visitDateLiteral(PartitionFilterParser.DateLiteralContext ctx) {
     PartitionFilterParser.DateContext date = ctx.date();
     String dateValue = unquoteString(date.value.getText());
     try {
-      return MetaStoreUtils.convertStringToDate(dateValue);
+      MetaStoreUtils.convertStringToDate(dateValue);
```

Comment:
are you mutating the input object?

Reply:
`MetaStoreUtils.convertStringToDate(dateValue)` converts to the date with time zone, it causes trouble when comparing the value both on direct sql or jdo, and as far as I know, there is no way to convert it back to original string, I would prefer to push the convention down into the `TreeVisitor`(i.e, the `JDOFilterGenerator`), converting it to date type if necessary.

Reply:
why do we need to validate then date here?

Reply:
We can avoid passing the invalid date down into the query filter generator given the context here is a date, and fail from the error quickly.